mmeds/server.py

```diff
@@ -130,9 +130,24 @@ def load_webpage(self, page, **kwargs):
         if necessary as well as any formatting arguments.
         """
         cp.log("Loading webpage")
+
         try:
             path, header = HTML_PAGES[page]
 
+            if header:
+                # Check if password needs to be reset
+                if kwargs.get('user') is not None:
+                    user = kwargs.get('user')
+                else:
+                    user = self.get_user()
+                with Database(owner=user, testing=self.testing) as db:
+                    reset_needed = db.get_reset_needed()
+                    if reset_needed:
+                        page = 'auth_change_password'
+                        kwargs['error'] = [(
+                            'Password change required. Your temporary password has been emailed to you.')]
+                        path, header = HTML_PAGES[page]
```

Comment:
I would think this check would only need to be performed when signing in. Having it here means it'll be performed for every page loading which will slow things down. Are there other situations I'm missing where it could come up?